From 8130eca321398eeda58253187a788ef18e0a5861 Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Sat, 27 Sep 2014 22:34:58 +0800
Subject: [PATCH 0248/1691] MLK-9617-10 usb: common: otg: update condition from
 a_host to a_suspend

commit 8130eca321398eeda58253187a788ef18e0a5861 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

This patch updates condition from a_host to a_suspend according to On-The-Go
and Embedded Host Supplement to the USB Revision 2.0 Specification Revision
2.0 version 1.1a.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit f3acf74150683d4555c82ad32b9c669d8a0a7fb5)
(cherry picked from commit a938eb1ccabe9213c0b232d45da2caeb75f8b9fe)
---
 drivers/usb/common/usb-otg-fsm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index eb1175b..2045ca9 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -334,8 +334,7 @@ int otg_statemachine(struct otg_fsm *fsm)
 	case OTG_STATE_A_HOST:
 		if (fsm->id || fsm->a_bus_drop)
 			otg_set_state(fsm, OTG_STATE_A_WAIT_VFALL);
-		else if ((!fsm->a_bus_req || fsm->a_suspend_req_inf) &&
-				fsm->otg->host->b_hnp_enable)
+		else if (!fsm->a_bus_req || fsm->a_suspend_req_inf)
 			otg_set_state(fsm, OTG_STATE_A_SUSPEND);
 		else if (!fsm->b_conn)
 			otg_set_state(fsm, OTG_STATE_A_WAIT_BCON);
-- 
1.9.1

